pandas/core/reshape/reshape.py

```diff
@@ -689,7 +689,7 @@ def _convert_level_number(level_num, columns):
         new_labels = [np.arange(N).repeat(levsize)]
         new_names = [this.index.name]  # something better?
 
-    new_levels.append(level_vals)
+    new_levels.append(frame.columns.levels[level_num])
```

Comment:
This is an actual change in the code. Is this to fix a bug? If so, could you do this as a separate PR (and add a test for it)?